internal/storage/storage.go

```diff
@@ -268,11 +271,49 @@ func (st *Storage) DownloadModule(toDir string, module *Module, progress Progres
 			continue
 		}
 		onlyLocalNoCopyArtifacts = false
-		if err = downloadArtifact(filepath.Join(toDir, sa.FileName), sa, callback, serverCert, retryCount,
-			retryInterval, st.done); err != nil {
+		var postProcess func(fileName string) error
+		if module.Metadata != nil && module.Metadata["AES256.key"] != "" {
+			var iv string
+			if iv = module.Metadata["AES256.iv"]; iv == "" {
+				return errors.New("AES256 key is provided, but initialization vector is missing. Only CBC encryption is supported")
+			}
+			postProcess = func(fileName string) error {
+				data, ppError := os.ReadFile(fileName)
```

Comment:
This will load the whole file in memory. Is it easy to mitigate this?

Reply:
There is no easy way to do this. There are a couple of ways for large file decryption, but the depends on how the file is encrypted. For now we could just point that encryption of large artifacts is discouraged and see what we can do about it in a separate issue.